libtiledbvcf/src/vcf/vcf_v3.cc

```diff
@@ -0,0 +1,397 @@
+/**
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2018 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "vcf/vcf_v3.h"
+
+namespace tiledb {
+namespace vcf {
+
+VCFV3::VCFV3()
```

Comment:
The VCFV3 is identical to the `VCF` with the following changes:
- `buffer_` is removed
- `buffer_offset_` is removed
- `record_queue_` is new
- `record_queue_pool_` is new
- `destroy_buffer` and `clear_buffer` have been removed
- `next` interface and implementation have been changed
- `return_record` is new
- `curr_record` is removed
- `read_records` implementation is changed

other misc stuff like `swap` and list initializer in the constructor
- `contig_name` accepts a record as an argument instead of using the `curr_record()`